bgpd/bgp_updgrp_packet.c

```diff
@@ -761,7 +761,7 @@ struct bpacket *subgroup_update_packet(struct update_subgroup *subgrp)
 				/* Flush the FIFO update queue */
 				while (adv)
 					adv = bgp_advertise_clean_subgroup(
-						subgrp, adj);
+						subgrp, adv->adj);
```

Comment:
This makes no sense to me.  adv->adj is the same value as adj( see line 696 ).  What path can we get to this point where adj is not adv->adj?  Furthermore why would we ever have a null adj value?  That seems like that is the actual problem.

Reply:
IMO, @louis-6wind is right (despite not being sure how adj comes as NULL), but we should reuse a new value of `adv` instead of the previous loop. Regarding `adj` as NULL, is this happening on UPDATE or WITHDRAW of prefixes? How can it be verified? In other words, @louis-6wind how did you get this crash?

Reply:
I'd really like to understand how this pointer is becoming NULL.  That is the real problem, imo.

Reply:
> I'd really like to understand how this pointer is becoming NULL. That is the real problem, imo.

![Screenshot-2023-05-30-22:05:47](https://github.com/FRRouting/frr/assets/3352707/3c769540-2eb9-4cc4-96a9-40d19153b495)

Reply:
again this makes no sense since adv is removed from the fifo.  How is it null and still on the fifo?

Reply:
to further clarify, the fact that we have a adv that is null that we want to use *implies* that we have the same adj multiple times on the baa.(is this legal?)  This seems like a fundamental mistake that should be fixed elsewhere instead of here as that we have violated some other principal.  Again I ask for a better understanding of how we got into this mess.